src/mqt/bench/devices/__init__.py

```diff
@@ -6,131 +6,206 @@
+        get_ionq_target("ionq_aria1"),
+        get_iqm_target("iqm_adonis"),
+        get_iqm_target("iqm_apollo"),
+        get_quantinuum_target("quantinuum_h2"),
+        get_rigetti_target("rigetti_aspen_m3"),
     ]
 
 
 @cache
 def get_available_device_names() -> list[str]:
-    """Get a list of all available device names."""
-    return [device.name for device in get_available_devices()]
+    """Return a list of available device names."""
+    return [device.description for device in get_available_devices()]
 
 
 @cache
-def _device_map() -> dict[str, Device]:
-    """One-time build of name → Device map.
+def _device_map() -> dict[str, Target]:
+    """Return a mapping of device names to Target objects."""
+    return {d.description: d for d in get_available_devices()}
 
-    Cached forever by functools.cache.
-    """
-    return {d.name: d for d in get_available_devices()}
 
-
-def get_device_by_name(device_name: str) -> Device:
-    """Get a device by its name.
-
-    Arguments:
-        device_name: the name of the device
-    """
+def get_device_by_name(device_name: str) -> Target:
+    """Return the Target object for a given device name."""
     try:
         return _device_map()[device_name]
     except KeyError:
-        msg = f"Device {device_name} not found in available devices."
+        msg = f"Device {device_name} not found."
         raise ValueError(msg) from None
 
 
-__all__ = [
-    "Device",
-    "DeviceCalibration",
-    "Gateset",
-    "get_available_device_names",
-    "get_available_devices",
-    "get_device_by_name",
-]
+def create_clifford_target(num_qubits: int) -> Target:
+    """Create a dense Target for the Clifford+T gateset with full qubit connectivity."""
+    gate_classes = {
+        "i": IGate,
+        "x": XGate,
+        "y": YGate,
+        "z": ZGate,
+        "h": HGate,
+        "s": SGate,
+        "sdg": SdgGate,
+        "t": TGate,
+        "tdg": TdgGate,
+        "sx": SXGate,
+        "sxdg": SXdgGate,
+        "cx": CXGate,
+        "cy": CYGate,
+        "cz": CZGate,
+        "swap": SwapGate,
+        "iswap": iSwapGate,
+        "dcx": DCXGate,
+        "ecr": ECRGate,
+        "measure": Measure,
+        "barrier": lambda: Barrier(num_qubits),
+    }
+
+    target = Target(num_qubits=num_qubits, description="clifford+t")
+
+    for gate_class in gate_classes.values():
+        try:
+            gate = gate_class()
+
+            if gate.name == "barrier":
+                qargs = {tuple(range(num_qubits)): None}
+            elif gate.num_qubits == 1:
+                qargs = {(q,): None for q in range(num_qubits)}
+            elif gate.num_qubits == 2:
+                qargs = {(q0, q1): None for q0 in range(num_qubits) for q1 in range(num_qubits) if q0 != q1}
+            else:
+                continue  # skip 3+ qubit gates (not needed here)
+
+            target.add_instruction(gate, qargs)
+        except Exception:
+            continue
```

Comment:
This can be much simpler I believe.
As per the docstring of the `Target.add_instruction` method, there is no need to supply properties if there aren't any.
So you also do not need the dictionary on top.
Simply call `target.add_instruction(Gate())` for each of the items in the list and that is it.